packages/react-ui-core/src/Gmap/GmapBase.js

```diff
@@ -1,32 +1,105 @@
-import React, { PureComponent } from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import { withScriptjs, withGoogleMap, GoogleMap } from 'react-google-maps'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import startsWith from 'lodash/startsWith'
+import pickBy from 'lodash/pickBy'
+import { MapContext } from './MapContext'
 
 const DEFAULT_ZOOM = 8
 const DEFAULT_CENTER = { // Atlanta, GA
   lat: 33.7490,
   lng: -84.3880,
 }
 
-export class BaseGoogleMap extends PureComponent {
+@themed(/^Gmap/, { pure: true })
+
+export default class GmapBase extends Component {
   static propTypes = {
+    theme: PropTypes.object,
+    className: PropTypes.string,
     children: PropTypes.node,
+    center: PropTypes.shape({
+      lat: PropTypes.number,
+      lng: PropTypes.number,
+    }),
+    zoom: PropTypes.number,
+    clickableIcons: PropTypes.bool,
+    mapControls: PropTypes.shape({
+      fullscreenControl: PropTypes.bool,
+      mapTypeControl: PropTypes.bool,
+      zoomControl: PropTypes.bool,
+      streetViewControl: PropTypes.bool,
+    }),
+  }
+
+  static defaultProps = {
+    theme: {},
+    zoom: DEFAULT_ZOOM,
+    clickableIcons: false,
+    mapControls: {
+      fullscreenControl: false,
+      streetViewControl: false,
+    },
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      map: null,
+    }
+    this.googleMap = React.createRef()
+    this.initMap = this.initMap.bind(this)
+  }
+
+  componentDidMount() {
+    this.initMap()
+  }
+
+  initMap() {
+    const {
+      zoom,
+      mapControls,
+      center,
+      clickableIcons,
+    } = this.props
+
+    const map = new window.google.maps.Map(this.googleMap.current, {
```

Comment:
Any chance of `window.google.maps` being null here?

Reply:
No chance because `initMap` doesn't get called until the component has mounted. The component won't mount until the google script has loaded.